addons/common/init_perFrameHandler.sqf

```diff
@@ -56,7 +56,10 @@ GVAR(waitUntilAndExecArray) = [];
         _delete = false;
     };
 
-
+    if (diag_frameno != GVAR(nextFrameNo)) then { // in case we skipped a frame, ensure we don't add to nextFrameBufferA while iterating
```

Comment:
this should be moved up to line 22 to ensure execNextFrame is also accurate for pfeh and waitAndExec